docs/Dialects/Verif.md

```diff
@@ -4,6 +4,322 @@ This dialect provides a collection of operations to express various verification
 
 [TOC]
 
+## Contracts
+
+Formal contracts are a key building block of the Verif dialect to help make formal verification scale to larger designs and deep module hierarchies.
+Contracts describe what a circuit expects from its inputs (`verif.require`) and what it guarantees its output to be (`verif.ensure`).
+The `verif.contract` op can be inserted into an SSA edge similar to an `hw.wire`, where the contract simply passes its operands through to its results.
+During formal verification the contract results are replaced by symbolic values that uphold the guarantees described in the contract's body.
+
+These contracts can then be used in two key ways:
+
+1.  A contract can be _verified_ to hold by turning `require`s into `assume`s and `ensure`s into `assert`s.
```

Comment:
I guess here you're not really verifying the contract, but you're checking that your region satisfies the contract, or that your region is valid under the contract. This follows the wording from the original paper this is based on.